server/index.js

```diff
@@ -26,15 +26,42 @@ app.get('/api/products', (req, res, next) => {
     "price",
     "image",
     "shortDescription"
-    from "products"
+  from "products"
   `;
 
   db.query(SQL)
     .then(result => {
       const data = result.rows;
       res.status(200).json(data);
     })
+    .catch(err => next(err));
+});
+
+app.get('/api/products/:productId', (req, res, next) => {
+  const productId = parseInt(req.params.productId);
+  const values = [productId];
+
+  if (isNaN(productId) || productId < 0) {
+    res.status(400).json({ error: 'Please enter positive number.' });
+    return;
+  }
 
+  const text = `
+  SELECT *
+  FROM "products"
+  WHERE "productId" = $1
+  `;
+
+  db.query(text, values)
+    .then(result => {
+      const data = result.rows[0];
+      if (!data) {
+        res.status(404).json({ error: '404 error' });
```

Comment:
You should be more descriptive in your error message. Also, I recommend passing a `new ClientError` to `next()` instead of responding in this part of your code. The error handler middleware at the bottom will do the responding for you.